libcudacxx/include/cuda/std/__atomic/types/small.h

```diff
@@ -63,9 +63,15 @@ _CCCL_HOST_DEVICE inline __atomic_small_proxy_t<_Tp> __atomic_small_to_32(_Tp __
 template <class _Tp, enable_if_t<!is_arithmetic_v<_Tp>, int> = 0>
 _CCCL_HOST_DEVICE inline _Tp __atomic_small_from_32(__atomic_small_proxy_t<_Tp> __val)
 {
+  _CCCL_DIAG_PUSH
+#if _CCCL_COMPILER(GCC, >=, 8)
+  // GCC starting with GCC8 warns about our extended floating point types having protected data members
+  _CCCL_DIAG_SUPPRESS_GCC("-Wclass-memaccess")
+#endif
```

Comment:
Paranoia: NVHPC has issues with warning suppression within classes 

Can we pull that diagnostic out of the function?

Reply:
Moved outside of function scope.